synapse/lib/ast.py

```diff
@@ -3086,16 +3098,71 @@ async def compute(self, runt, path):
         if self.isconst:
             return self.constval
 
+        if not isinstance(self.kids[0], Const):
+            valu = await self.kids[0].compute(runt, path)
+            valu = await s_stormtypes.toprim(valu)
+
+            if not isinstance(valu, str):
+                mesg = 'Invalid value type for tag name, tag names must be strings.'
+                raise s_exc.BadTypeValu(mesg=mesg)
+
+            normtupl = await runt.snap.getTagNorm(valu)
+            return normtupl[0]
+
         vals = []
         for kid in self.kids:
             part = await kid.compute(runt, path)
             if part is None:
                 mesg = f'Null value from var ${kid.name} is not allowed in tag names.'
                 raise kid.addExcInfo(s_exc.BadTypeValu(mesg=mesg))
-            vals.append(await tostr(part))
+
+            part = await tostr(part)
+            partnorm = await runt.snap.getTagNorm(part)
+            vals.append(partnorm[0])
 
         return '.'.join(vals)
 
+    async def compute_array(self, runt, path, excignore=()):
```

Comment:
Since this behavior seems specific to tags ( or is it? ) perhaps we should name this something more like `computeTagArray()` ?